.github/workflows/ci.yml

```diff
@@ -0,0 +1,110 @@
+name: CI
+
+on:
+  pull_request:
+
+env:
+  node_version: 20
+  pnpm_version: 8
+
+jobs:
+  format:
```

Comment:
I suggest to just run the lint action and prevent a PR if that fails. This puts the onus of formatting on the PR submitting party and doesn't automatically format end maybe accidentally breaking things. The important part of this strategy is to actually prevent unformatted PRs from being merged. Not sure how to do that with GitHub as I only do Azure Pipelines for work and the coniguration is different.

Reply:
Nice catch! Updated to let `lint` handle format checking and not run format beforehand.